### PR TITLE
Shrink main func to resolve gocyclo warning

### LINKED ISSUE #178: Fix cyclomatic complexity in examples/simple-udp/server/main.go
Remove the // nolint comment.

The main function has a cyclomatic complexity greater than 10. Refactor the function to lower it.

### PR DESCRIPTION
**Pull request by enocom**
Fixes #178 

This is the simplest fix to the gocyclo linter error: breaking out functions of related code. I propose taking the same approach for #177.

### CODE REVIEW

File: examples/simple-udp/server/main.go [hunk 1]
```diff
@@ -50,7 +49,7 @@ func main() { // nolint: gocyclo
 	}
 
 	log.Print("Starting Health Ping")
-	stop := make(chan bool)
+	stop := make(chan struct{})
```
Comment (markmandel) [inline examples/simple-udp/server/main.go]: Just for my own curiosity - why the change? Because it more closely matches `context.Context` and some of the Kubernetes constructs?

Comment (enocom) [reply examples/simple-udp/server/main.go]: Since `stop` is used purely as a signally mechanism, and not as a way to transport meaningful information, I have followed the Go idiom of using `struct{}`, which indicates the value is of no meaning in and of itself. On the other hand, `bool` suggests we might send `true` or `false`, when in fact we don't really care about either.

Comment (markmandel) [reply examples/simple-udp/server/main.go]: Thanks! That makes perfect sense!
